components/blitz/src/omero/gateway/facility/BrowseFacility.java

```diff
@@ -240,6 +260,29 @@ public IObject findIObject(SecurityContext ctx, String klassName, long id)
      *
      * @param ctx
      *            The security context.
+     * @param pojoName
+     *            The type of object to retrieve. (Either the simple or the full
+     *            class name, e. g. omero.gateway.model.DatasetData or
+     *            DatasetData)
+     * @param id
+     *            The object's id.
+     * @return The last version of the object.
+     * @throws DSOutOfServiceException
+     *             If the connection is broken, or not logged in
+     * @throws DSAccessException
+     *             If an error occurred while trying to retrieve data from OMERO
+     *             service.
+     */
+    public DataObject findObject(SecurityContext ctx, String pojoName, long id)
```

Comment:
Would a generic version of this be easier to use in general, or would that be too constraining based on what the client knows at a given point? e.g. `public <T extends DataObject> T findObject(Class<T> pojoType, ...)` would be an option.

The same holds for `findIObject`.